tikv-client/src/main/java/com/pingcap/tikv/PDClient.java

```diff
@@ -95,34 +97,93 @@ public class PDClient extends AbstractGRPCClient<PDBlockingStub, PDStub>
   private List<URI> pdAddrs;
   private Client etcdClient;
   private ConcurrentMap<Long, Double> tiflashReplicaMap;
+  private final org.tikv.common.ReadOnlyPDClient upstreamPDClient;
 
-  private PDClient(TiConfiguration conf, ChannelFactory channelFactory) {
+  private PDClient(TiConfiguration conf, RequestKeyCodec keyCodec, ChannelFactory channelFactory) {
     super(conf, channelFactory);
     initCluster();
     this.blockingStub = getBlockingStub();
     this.asyncStub = getAsyncStub();
+    this.upstreamPDClient = createUpstreamPDClient(keyCodec);
+  }
+
+  private org.tikv.common.ReadOnlyPDClient createUpstreamPDClient(RequestKeyCodec keyCodec) {
+    org.tikv.common.TiConfiguration tikvConf = convertTiConfiguration(conf);
+    org.tikv.common.util.ChannelFactory channelFactory = convertChannelFactory(conf, tikvConf);
+    return org.tikv.common.PDClient.create(tikvConf, keyCodec, channelFactory);
+  }
+
+  private org.tikv.common.TiConfiguration convertTiConfiguration(
```

Comment:
Should be careful about this convertation.
Can you give a doc about diff and mapping of two configurations

Reply:
Maybe a conversion tool class code can more clearly show how we convert some upstream and downstream configurations. as I am
[tikv-client/src/main/java/com/pingcap/tikv/util/ConverterUpstream.java](https://github.com/pingcap/tispark/pull/2399/files#diff-127d716c9af974c920da0c8817c800a1b7029895c7fd0514eae3ac454d12c3b5:~:text=tikv%2Dclient/src/main/java/com/pingcap/tikv/util/ConverterUpstream.java)
  as in the file